frame/node-authorization/src/lib.rs

```diff
@@ -0,0 +1,761 @@
+            ensure!(Owners::<T>::get(&node) == sender, Error::<T>::NotOwner);
+
+            Owners::<T>::insert(&node, &owner);
+
+            Self::deposit_event(RawEvent::NodeTransferred(node, sender, owner));
+        }
+
+        /// Add additional connections to a given node.
+        ///
+        /// - `node`: identifier of the node.
+        /// - `connections`: additonal nodes from which the connections are allowed.
+        #[weight = T::WeightInfo::add_connections()]
+        pub fn add_connections(
+            origin,
+            node: NodePublicKey,
+            connections: Vec<NodePublicKey>
+        ) {
+            let sender = ensure_signed(origin)?;
+            ensure!(Owners::<T>::contains_key(&node), Error::<T>::NotClaimed);
+            ensure!(Owners::<T>::get(&node) == sender, Error::<T>::NotOwner);
+
+            let mut nodes = AdditionalConnections::get(&node);
+
+            for add_node in connections.iter() {
+                if *add_node == node {
+                    continue;
+                }
+                if let Err(add_location) = nodes.binary_search(add_node) {
+                    nodes.insert(add_location, add_node.clone());
+                }
+            }
+            
+            AdditionalConnections::insert(&node, nodes);
+
+            Self::deposit_event(RawEvent::ConnectionsAdded(node, connections));
+        }
+
+        /// Remove additional connections of a given node.
+        ///
+        /// - `node`: identifier of the node.
+        /// - `connections`: additonal nodes from which the connections are not allowed anymore.
+        #[weight = T::WeightInfo::remove_connections()]
+        pub fn remove_connections(
+            origin,
+            node: NodePublicKey,
+            connections: Vec<NodePublicKey>
+        ) {
+            let sender = ensure_signed(origin)?;
+            ensure!(Owners::<T>::contains_key(&node), Error::<T>::NotClaimed);
+            ensure!(Owners::<T>::get(&node) == sender, Error::<T>::NotOwner);
+
+            let mut nodes = AdditionalConnections::get(&node);
+
+            for remove_node in connections.iter() {
+                if let Ok(remove_location) = nodes.binary_search(remove_node) {
+                    nodes.remove(remove_location);
+                }
+            }
+            
+            AdditionalConnections::insert(&node, nodes);
+
+            Self::deposit_event(RawEvent::ConnectionsRemoved(node, connections));
+        }
+
+        /// Set reserved node every block. If may not be enabled depends on the offchain
+        /// worker CLI flag.
+        fn offchain_worker(now: T::BlockNumber) {
+            let node_public_key = sp_io::offchain::get_node_public_key();
+            match node_public_key {
+                Err(_) => debug::error!("Error: failed to get public key of node at {:?}", now),
+                Ok(node) => sp_io::offchain::set_reserved_nodes(
```

Comment:
The only drawback of this API I see is that initially when you start the node it's not going to be configured to connect to reserved peers, it only changes after first block is imported (or rather first offchain worker is run).

Is there a way to make this persistent or maybe somehow re-run OCW for latest block on startup?

Reply:
Yeah, make the reserved nodes persistent is in plan.  I guess it's possible to do a one-run OCW  in `build_offchain_workers` for latest block when restarting the node, but it's not very useful when the node is not in sync, in this scenario, I guess it's better to disable OCW with manually reserving nodes.